src/npf/rproc/npf_ext_appfw.c

```diff
@@ -389,10 +389,15 @@ appfw_action(npf_cache_t *npc, struct rte_mbuf **nbuf, void *arg,
 	 */
 	dpi_flow = npf_session_get_dpi(se);
 	if (!dpi_flow) {
+#ifdef USE_NDPI
 		uint8_t engines[] = { IANA_USER, IANA_NDPI };
-
+		size_t engines_len = 2;
+#else
+		uint8_t engines[] = { IANA_USER };
+		size_t engines_len = 1;
+#endif /* USER_NDPI */
```

Comment:
@dfawcus

Not directly related, but possibly important.

I believe that when the app-fw was created, 0 data-length packets were not fed through it, and so did not count towards the number of initial packets it had to see before converging on an answer. So TCP flags only updates would not count.

The context of this PR implies that this is no longer the case, and it may inform upon why we had issues with the initial count behaviour in the post Qosmos world. So there may be something to be updated in the app-fw now as to which packets it counts towards that initial set - possibly 0-data length TCP packet should not be counted?

Reply:
Possibly copied to the wrong place, this makes more sense in relation to the prior commit.
(I can't recall which commit I added the comment to in the internal PR system).